lib/workspace.py

```diff
@@ -143,7 +143,8 @@ def update(self):
 
                 # 4. If the repo has a selected version, fail if that version
                 # does not include this tuple's hash
-                if not repo.is_ancestor(commit, selected_commit):
+                if not (repo.is_ancestor(commit, selected_commit)
+                        or repo.is_ancestor(selected_commit, commit)):
```

Comment:
We need to run some special code in this scenario.

> If the author of bar:2018 explicitly specified xyz:2010 despite newer versions existing (which is the only way that foo:2017 could have a newer version: xyz:2011), is it safe to override that requirement with the newer version?

I think a warning would be a good thing to put here.

We also need to update the package to the newer version requested. When we do this, we might leave behind a dependency no longer needed (not the end of the world, but still undesired behavior).